code/modules/projectiles/guns/projectile/shotgun.dm

```diff
@@ -143,6 +150,82 @@
 			if(istype(user, /mob/living/carbon/human) && src.loc == user)
 				var/mob/living/carbon/human/H = user
 				H.update_inv_hands()
+	if(istype(A, /obj/item/stack/rods))
+		to_chat(user, "<span class='notice'>You start jamming a rod into \the [src]'s barrels to try and loosen the shells...</span>")
+		if(do_after(user, src, 60))
+			jammed = 0
+			to_chat(user, "<span class='notice'>You manage to loosen the shells.</span>")
+
+/obj/item/weapon/gun/projectile/shotgun/doublebarrel/verb/toggle_doubleshot()
+	set name = "Toggle Shooting Both Barrels"
+	set category = "Object"
+	if((world.time >= last_fired + fire_delay) && !doubleshooting)
+		doubleshot = !doubleshot
+		if(!doubleshot)
+			fire_delay = initial(fire_delay)
+		to_chat(usr, "You switch \the [src]'s fire selector to [doubleshot ? "fire both barrels at once" : "fire one barrel at a time"].")
+
+#define JAMMED 1
+#define DROPGUN 2
+#define KNOCKDOWN 3
+#define RECOILBRUISE 4
+
+/obj/item/weapon/gun/projectile/shotgun/doublebarrel/Fire(atom/target as mob|obj|turf|area, mob/living/user as mob|obj, params, reflex = 0, struggle = 0)
+	if(doubleshot && (getLiveAmmo() == 2)) //ANGERY
+		var/atom/reverse = locate(2*user.x - target.x, 2*user.y - target.y, target.z)
+
+		if(ready_to_fire())
+			fire_delay = 0
+		else
+			return
+		if(defective && prob(5))
+			to_chat(user, "<span class='danger'>\The [src] can't handle the pressure of firing two shells at once!</span>")
+			explosion(get_turf(loc), -1, 0, 2)
+			user.drop_item(src, force_drop = 1)
+			qdel(src)
+		else
+			doubleshooting = 1
+			recoil = 2 * initial(recoil)
+			..()
+			..()
+			message_admins("[usr] just fired both barrels out of \his [src].")
+			fire_delay = 20
+			recoil = initial(recoil)
+			doubleshooting = 0
+			if(prob(recoileffectchance)) // Total chance to fuck up
+				var/mob/living/carbon/human/H = user
+				switch(pick(JAMMED,DROPGUN,KNOCKDOWN,RECOILBRUISE))
+					if(JAMMED)
+						jammed = 1 // Needs some work to be unloaded and reloaded again
+					if(DROPGUN)
+						if(user.drop_item(src)) // Launches it from your hands behind you, letting someone else steal it
+							src.throw_at(reverse, 2, 10)
+							to_chat(user, "<span class='danger'>The recoil is too strong and \the [src] flies out of your hand!</span>")
+						else
+							to_chat(user, "<span class='notice'>You barely manage to withstand the recoil.</span>")
+					if(KNOCKDOWN)
+						H.Stun(3) //Drops you on your ass, this is a death sentence if you're in combat
+						H.Knockdown(3)
+						to_chat(user, "<span class='danger'>The recoil throws you off balance!</span>")
+					if(RECOILBRUISE)
+						var/datum/organ/external/org = H.find_organ_by_grasp_index(user.is_holding_item(src)).parent // It should break your arm, not your hand, the gun still has a stock you're bracing against.
+						org.take_damage(rand(15,40), null , null, null)// I have no idea what the fuck I am doing, adjustBruteLossByPart() doesn't work for some reason.
```

Comment:
No need for the null, null, null chain. You can underload function arguments in byond.